### PR TITLE
Clean up add_payment

### PR DESCRIPTION
**Pull request by Krigpl**
So that TransactionSystem doesn't create `Payment` object

### CODE REVIEW

**Review by maaktweluit — APPROVED** [review]

File: golem/ethereum/paymentprocessor.py [hunk 1]
```diff
@@ -106,23 +112,22 @@ def _on_batch_confirmed(self, payments: List[Payment], receipt) -> None:
...
-            payment.subtask,
-            encode_hex(payment.payee),
-            payment.value / denoms.ether))
-
-        if not payment.processed_ts:
-            payment.processed_ts = get_timestamp()
-            payment.save()
+    def add(self, subtask_id: str, eth_addr: str, value: int) -> Payment:
+        log.info(
+            "Adding payment for %s to %s (value: %f)",
```
Comment (maaktweluit) [inline golem/ethereum/paymentprocessor.py:117]: `%.6f` formatter is removed, was this meant?

Comment (Krigpl) [reply golem/ethereum/paymentprocessor.py:117]: The default is `.6` anyway, so yeah

File: golem/ethereum/paymentprocessor.py [hunk 2]
```diff
@@ -106,23 +112,22 @@ def _on_batch_confirmed(self, payments: List[Payment], receipt) -> None:
...
+            subtask_id,
+            eth_addr,
+            value / denoms.ether,
+        )
+        payment = Payment.create(
+            subtask=subtask_id,
+            payee=decode_hex(eth_addr),
```
Comment (mfranciszkiewicz) [inline golem/ethereum/paymentprocessor.py:124]: Why remove the `len` check?

Comment (Krigpl) [reply golem/ethereum/paymentprocessor.py:124]: I'd say we either do full validation or none at all and we didn't validate subtask_id nor value.
These args are coming from SRA message and are validated there (afaik).

**Pull request merged**